src/functions.php

```diff
@@ -38,3 +40,37 @@ function usleep(int $time, LoopInterface $loop) : PromiseInterface
 {
     return Timer\resolve($time/1000000, $loop);
 }
+
+/**
+ * Returns the content type in MIME format, like text/plain or
+ * application/octet-stream, or rejects with a \RuntimeException on failure
+ *
+ * @param string $fileName
+ * @param LoopInterface $loop
+ *
+ * @return PromiseInterface
+ */
+function mime_content_type(string $fileName, LoopInterface $loop) : PromiseInterface
+{
+    $process = new Process(sprintf('php -d error_reporting=0 -r "echo mime_content_type(\'%s\');"', $fileName));
```

Comment:
Possible command injection vulnerability when file name contains special characters. Should probably use `escapeshellarg()` (or if you're feeling fancy some base64 encoding/decoding etc.).

Reply:
Thank you! I didn't even think about it 🙈 
@mmoreram maybe this also requires a separate test that we were thinking about?

Reply:
@clue Omg. You're right. Do we have any example for tests?

Reply:
Something like: `x\'); echo(mysql_bad_stuff()); echo ('x`

Reply:
Updated!